src/Umbraco.Web.UI/Views/Partials/BlockList/Default.cshtml

```diff
@@ -0,0 +1,20 @@
+﻿@inherits UmbracoViewPage<BlockListModel>
+@using Umbraco.Core.Models.Blocks
+@{
+    if (Model?.Layout == null || !Model.Layout.Any()) { return; }
+}
+<div class="umb-block-list">
+    @foreach (var layout in Model.Layout)
+    {
+        if (layout?.Udi == null) { continue; }
+        var data = layout.Data;
+        try
+        {
+            @Html.Partial("BlockList/Components/" + data.ContentType.Alias, (data, layout.Settings))
+        }
+        catch (Exception ex)

```

Comment:
Yes we should absolutely *not* have a try/catch here. Things to note about catching errors:

* You should never catch the generic Exception type, you should always catch a real Exception type that you expect to have happen. When you catch a generic Exception it means you don't know what might happen which means you shouldn't try/catch in the first place. In some cases if you are catching a generic Exception class you should comment as to why.
* Never try/catch/swallow without explicit comments of why you are doing that
* Never try/catch 'just in case' and you can almost always avoid try/catch with proper error checking

All this is going to do is:

* Make the view more complex
* Hide errors from being resolve
* Make it confusing for the end user and developer as to why the data isn't showing

Reply:
Totally agreed are you able to rectify @benjaminc, please